sci-physics/yoda/yoda-1.9.9.ebuild

```diff
@@ -0,0 +1,88 @@
+# Copyright 2022-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit bash-completion-r1 autotools python-single-r1 optfeature
+
+DESCRIPTION="Yet more Objects for (High Energy Physics) Data Analysis"
+HOMEPAGE="https://yoda.hepforge.org/"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://gitlab.com/hepcedar/yoda"
+else
+	SRC_URI="https://yoda.hepforge.org/downloads?f=${P^^}.tar.bz2 -> ${P^^}.tar.bz2"
+	S="${WORKDIR}/${P^^}"
+	KEYWORDS="~amd64"
+fi
+
+LICENSE="GPL-3"
+SLOT="0/${PV}"
+IUSE="root test +python +zlib"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} ) root? ( python )"
+
+RDEPEND="
+	root? ( sci-physics/root:=[${PYTHON_SINGLE_USEDEP}] )
+	python? ( ${PYTHON_DEPS} )
+	zlib? ( sys-libs/zlib )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	python? (
+		$(python_gen_cond_dep '
+			>=dev-python/cython-0.29.24[${PYTHON_USEDEP}]
+		')
+		test? (
+			$(python_gen_cond_dep '
+				dev-python/numpy[${PYTHON_USEDEP}]
+				dev-python/matplotlib[${PYTHON_USEDEP}]
+			')
+		)
+	)
+"
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	default
+	# reconf due to python3_10 patch
+	eautoreconf
+}
+
+src_configure() {
+	# we need to use the prefix cython here
+	econf --disable-static \
+		$(use_enable root) \
+		$(use_enable python pyext) \
+		$(use_with zlib zlib "${ESYSROOT}/usr") \
+		$(usex python CYTHON="${ESYSROOT}/usr/bin/cython")
+}
+
+src_test() {
+	# PYTESTS and SHTESTS both require python tools
+	if use python; then
+		emake check
+	else
+		emake check PYTESTS= SHTESTS= NO_PYTHON=1
+	fi
+}
```

Comment:
For the current `::gentoo` ebuild I get:
```
============================================================================
Testsuite summary for YODA 1.9.7
============================================================================
# TOTAL: 21
# PASS:  21
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
```

But the new version gives me:
```
============================================================================
Testsuite summary for YODA 1.9.10
============================================================================
# TOTAL: 43
# PASS:  21
# SKIP:  0
# XFAIL: 0
# FAIL:  22
# XPASS: 0
# ERROR: 0
```

More tests are run, but they fail with:

```
ModuleNotFoundError: No module named 'yoda'
```

Probably what we want here is to set `PYTHONPATH="${S}"` or something similar.

Reply:
Can you send me what use flags and python target you used? My 1.9.9 and 1.9.10 does not fail and I X-checked that I do not have yoda installed. Is it maybe only `python3_12` that fails for you for some reason. I guess it is not numpy or mpl missing. they are in the `BDEPEND` either way.

Reply:
Though I get the bash completion warnings on my desktop (in contrast to my CI)

Reply:
Here's the build log and test log:
[yoda-1.9.10:20240315-183716.txt](https://github.com/gentoo/gentoo/files/14619533/yoda-1.9.10.20240315-183716.txt)
[test-suite.txt](https://github.com/gentoo/gentoo/files/14619538/test-suite.txt)

Reply:
Jumps right to doxy for you 
```
configure: YODA Python lib will be installed to /usr/lib/python3.12/site-packages
./configure: 19022: test: xyes: unexpected operator
./configure: 19033: test: xyes: unexpected operator
checking for doxygen... /usr/bin/doxygen
```
in my case it just accepts pyext and configures it before doxygen

So it seems to be `test x$enable_pyext` that complains, is it maybe some difference between m4 versions. I am also still on gcc-12.

Reply:
NVW, my CI also used GCC 13 while I used 12.

Reply:
they configure with `test x$enable_pyext == xyes` when it should be `test "x$enable_pyext" = xyes`. Which shell are you using that is sensitive of this case? I think it should just be one equal sign quotes dont matter here...

Reply:
> they configure with `test x$enable_pyext == xyes` when it should be `test "x$enable_pyext" = xyes`. Which shell are you using that is sensitive of this case? I think it should just be one equal sign quotes dont matter here...

Main (user) shell is zsh, /bin/sh is linked to dash. This could be the cause of the problem, dash is a bit stricter POSIX.

Reply:
I added a patch that should fix it. Could you give it another shot at some point?

Reply:
Just tested and now it works on my end too :partying_face: 

Warning about the bash completions remains, but perhaps it makes sense to fix this upstream instead of providing our own completions.